File: viewbindingdemo/src/main/java/com/example/viewbindingdemo/MainActivity.java
package com.example.viewbindingdemo;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.viewbindingdemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = this;
//        setContentView(R.layout.activity_main);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.tvName.setText("Hello ViewBinding");

        binding.ivAvatar.setImageResource(R.mipmap.ic_launcher);

        binding.btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(context, "hello world", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: navigationdemo/src/main/java/com/example/navigationdemo/MainActivity.kt
package com.example.navigationdemo

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.navigationdemo.login.WelcomeActivity
import com.example.navigationdemo.tab.TabActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun click1(view: View) {
        startActivity(Intent(this@MainActivity, WelcomeActivity::class.java))
    }

    fun click2(View: View) {
        startActivity(Intent(this@MainActivity, TabActivity::class.java))
    }

}